docs: Improve wording for labels and services policies (#27171)
cilium-team/cilium: Documentation/security/policy/language.rst

The layer 3 policy establishes the base connectivity rules regarding which endpoints
can talk to each other. Layer 3 policies can be specified using the following methods:

* `Endpoints based`: This is used to describe the relationship if both

>   endpoints are managed by Cilium and are thus assigned labels. The big

Reviewer: The adjective is subjective.
```suggestion
  endpoints are managed by Cilium and are thus assigned labels. The
```

  above. DNS information is acquired by routing DNS traffic via a proxy.
  DNS TTLs are respected.

.. _Endpoints based:

Endpoints Based
---------------

Label-based L3 policy is used to establish policy between endpoints inside the

> cluster managed by Cilium. Label-based L3 policies are defined by using an

Reviewer 2: There are two more references to `label-based` here, do these need to change as well?

Author: Ah good point, yes let me fix that up.

  above. DNS information is acquired by routing DNS traffic via a proxy.
  DNS TTLs are respected.

.. _Endpoints based:

Endpoints Based
---------------

> Endpoints-based L3 policy is used to establish policy between endpoints inside

Reviewer: "establish policy" is an awkward phrase and ambiguous.
```suggestion
Endpoints-based L3 policy is used to establish rules between endpoints inside
```

the cluster managed by Cilium. Endpoints-based L3 policies are defined by using

> an `EndpointSelector` inside a rule to choose what kind of traffic that can be

Reviewer: "choose" is not as specific as "select". "that" is extraneous.
```suggestion
an `EndpointSelector` inside a rule to select what kind of traffic can be
```

Author: Applied these changes, thanks for the suggestions.
